src/main/java/com/teragrep/pth_06/ArchiveMicroStreamReader.java

```diff
@@ -98,6 +103,15 @@ public final class ArchiveMicroStreamReader implements MicroBatchStream {
 
         this.config = config;
 
+        if (config.isHdfsEnabled) {
+            this.hq = new HdfsQueryProcessor(config);
+            hdfsOffsets = hq.hdfsOffsetMapToJSON();
+        }
+        else {
+            this.hq = null;
```

Comment:
These null assignments should be avoided. Perhaps implement a stub object? I'm aware that this was already present before this PR but it should be fixed at some point.

Reply:
Solved in commit c72378bd9aac4dda9403647c77c09cc64eb595af